eq_translations/survey_schema.py

```diff
@@ -1,245 +1,204 @@
+                    continue
 
-        return context_placeholder_pointers, no_context_placeholder_pointers
+                self.total_translatable_strings += 1
 
-    def get_message_pointers(self):
-        """
-        Message pointers need to be iterated as a dict and each key added individually
-        :return:
-        """
-        pointers = []
+                schema_element = resolve_pointer(self.schema, pointer)
+                if isinstance(schema_element, dict):
+                    yield from self._get_pointers_from_dict_element(
+                        pointer, schema_element, with_context
+                    )
 
-        message_pointers = find_pointers_to(self.schema, "messages")
-        for message_pointer in message_pointers:
-            schema_element = resolve_pointer(self.schema, message_pointer)
-            for element in schema_element:
-                pointers.append(f"{message_pointer}/{element}")
-        return pointers
+                elif isinstance(schema_element, list):
+                    yield from self._get_pointers_from_list_element(
+                        pointer, schema_element, with_context
+                    )
 
-    def get_list_pointers(self):
-        """
-        List pointers need to be iterated and each element added individually
-        :return:
-        """
-        pointers = []
+                else:
+                    yield self._get_pointer_dict(pointer, schema_element, with_context)
 
-        list_pointers = find_pointers_to(self.schema, "list")
-        for list_pointer in list_pointers:
-            schema_element = resolve_pointer(self.schema, list_pointer)
-            if isinstance(schema_element, list):
-                for i, p in enumerate(schema_element):
-                    # placeholders are being skipped as they are dealt with elsewhere
-                    if not is_placeholder(p):
-                        pointers.append(f"{list_pointer}/{i}")
+    def _get_pointers_from_dict_element(self, pointer, schema_element, with_context):
+        plural_forms = schema_element.get("text_plural")
+        if plural_forms:
+            new_pointer = f"{pointer}/text_plural"
+            yield self._get_pointer_dict(new_pointer, plural_forms, with_context)
 
-        return pointers
+        for element in schema_element:
+            value = schema_element[element]
+            if isinstance(value, str):
+                new_pointer = f"{pointer}/{element}"
+                yield self._get_pointer_dict(new_pointer, value, with_context)
 
-    def get_title_pointers(self):
-        """
-        Titles need to be handled separately as they may require context for translation
-        """
-        return find_pointers_to(self.schema, "title")
+    def _get_pointers_from_list_element(self, pointer, schema_element, with_context):
+        for index, _ in enumerate(schema_element):
+            value = schema_element[index]
+            if isinstance(value, str):
```

Comment:
When is `value` not a string?

Reply:
`value` could be a placeholder.

Reply:
We should check if the object is a dict and if so extract the `text` element. It works currently as there is only one string property in a placeholder, if another was added this would extract it as a translatable string.

Also, do we allow plurals in list elements?

Reply:
`text` is now extracted explicitly. Added plural support also.